tests/subsys/mpsl/pm/pm_test.c

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <unity.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+
+#include <errno.h>
+
+#include "cmock_policy.h"
+#include "cmock_mpsl_pm.h"
+#include "cmock_mpsl_pm_config.h"
+#include "cmock_mpsl_work.h"
+
+#include <mpsl/mpsl_pm_utils.h>
+#include <zephyr/kernel.h>
+
+#define NUMELTS(X) (sizeof((X)) / sizeof(X[0]))
+
+#define PM_MAX_LATENCY_HCI_COMMANDS_US 4999999
+#define MAX_DELAY_SINCE_READING_PARAMS 50
+
+#define TIME_TO_REGISTER_EVENT_IN_ZEPHYR 1000
+#define RETRY_TIME_MAX_US (UINT32_MAX - TIME_TO_REGISTER_EVENT_IN_ZEPHYR)
+
+/* Mock implementation for mpsl_work_q*/
+struct k_work_q mpsl_work_q;
+
+/* The unity_main is not declared in any header file. It is only defined in the generated test
+ * runner because of ncs' unity configuration. It is therefore declared here to avoid a compiler
+ * warning.
+ */
+extern int unity_main(void);
+typedef enum {
+	EVENT_FUNC_NONE,
+	EVENT_FUNC_REGISTER,
+	EVENT_FUNC_UPDATE,
+	EVENT_FUNC_UNREGISTER,
+	EVENT_FUNC_DELAY_SCHEDULING,
+} event_func_t;
+
+typedef enum {
+	LATENCY_FUNC_NONE,
+	LATENCY_FUNC_REGISTER,
+	LATENCY_FUNC_UPDATE,
+} latency_func_t;
+
+
+typedef struct {
+	bool new_test;
+	bool pm_params_get_retval;
+	mpsl_pm_params_t params;
+	event_func_t event_func;
+	uint64_t event_time_us;
+	latency_func_t latency_func;
+	uint64_t latency_us;
+} test_vector_t;
+
+void test_pm_with_table(void)
```

Comment:
```suggestion
void test_pm(void)
```
or break up in smaller functions. See below.